docs/design/deployment_design.md

```diff
@@ -0,0 +1,199 @@
+
+  ![Simple Endpoint](png/mizar-phase-1.png)
+
+- **Mizar Phase 2**: Mizar is targeting to 2-cluster model. In the model, Mizar is running at a k8s or Arktos cluster, and it provides network functions to another k8s or Arktos cluster through Mizar proxy.
+
+  Mizar cluster should be able to run either k8s or Arktos.
+
+  ![Simple Endpoint](png/mizar-phase-2.png)
+
+We should produce same major deploy process for Mizar phase 1 and 2, while deploying different components.
+
+### Environments
+
+There are different environments that Mizar should be supporting.
+
+- k8s Kind environment
+
+  Typically k8s cluster can be launched by Kind. It's k8s in one VM, but with multiple nodes.
+
+  Mizar phase 2 requires two clusters. If Kind doesn't support deploy two clusters to one VM, we can use other means such as arktos-up to deploy one cluster other than cluster deployed by Kind.
+
+- k8s GCP environment
+
+  kubernetes running in Google Cloud Platform.
+
+- k8s AWS environment
+
+  kubernetes running in AWS.
+
+- arktos-up environment
+
+  Arktos running in one machine in one node mode. It's typical Arktos environment for dev and test.
+
+- Arktos GCP environment - single master
+
+  Arktos running in GCP. It has one master node and multiple worker nodes.
+
+- Arktos AWS environment - single master
+
+  Arktos running in AWS. It has one master node and multiple worker nodes.
+
+- Arktos GCP environment - multi master
+
+  Arktos running in GCP. It has multiple API servers and multiple ETCD. Consider it has multiple master nodes.
+
+- Arktos GCP environment with Kubemark
+
+  Arktos running in GCP. It uses Kubemark to simulate large clusters with hundreds of nodes. It's a good environment to demonstrate Mizar performance in large clusters.
+
+### Dev vs Prod of Mizar
+
+All the time these should be two version of Mizar to be deployed. One is dev version and one is prod version.
+
+- **Dev Version**, it always reflects latest code of Mizar. The code can be from any branch such as master, dev-next or any private branch. In this way, while developing Mizar, developers can easily test Mizar by latest code. It's for Mizar developers with dev or test purpose. 
+
+- **Prod Version**, it's for end users to use Mizar in their test or prod environment. Its purpose is not to test Mizar. Instead it's to test their environment with Mizar, or check how Mizar fit their requirements, or using Mizar in their production environment. It's to deploy Mizar stable versions. 
+
+	We shall provide public docker images for prod version deployment. 
+
+### New Install vs Update
+
+The deploy script should support both new install and update of Mizar. In prod environment, when there is Mizar installed, user may want to update Mizar to new version while cluster networking won't break. We need to support "Mizar update" scenario. While Mizar updating, there should be no impact or minimal/acceptable impact to the cluster.
```

Comment:
I would consider a CI/CD pipeline for these as well! In both prod/test environment, it is a matter of branch selection.

Reply:
Updated.